TYPO3.Flow/Classes/TYPO3/Flow/Http/Request.php

```diff
@@ -373,6 +417,136 @@ public function getContent($asResource = false)
     }
 
     /**
+     * Set a trusted proxy header.
+     *
+     * This is primarily useful for testing. If you want to set trusted headers, use the configuration setting
+     * "TYPO3.Flow.http.trustedProxies.headers" instead.
+     *
+     * @param string $type One of the HEADER_* constants
+     * @param string $name The header name to trust or an empty string to not trust this header
+     * @return void
+     */
+    public static function setTrustedProxyHeader($type, $name)
+    {
+        if (!in_array($type, array(self::HEADER_CLIENT_IP, self::HEADER_HOST, self::HEADER_PORT, self::HEADER_PROTOCOL))) {
+            return;
+        }
+        self::$trustedProxiesSettings['headers'][$type] = strval($name);
+    }
+
+    /**
+     * Set the trusted proxies IP addresses.
+     * If the argument is an empty array, no proxies are trusted.
+     * If the argument is the string '*', all proxies are trusted.
+     *
+     * This is primarily useful for testing. If you want to set trusted headers, use the configuration setting
+     * "TYPO3.Flow.http.trustedProxies.proxies" instead.
+     *
+     * @param array|string $trustedProxies
+     * @return void
+     */
+    public static function setTrustedProxies($trustedProxies)
+    {
```

Comment:
Fair enough. We made some bad experiences with mixed argument types though, as it makes the code more complex and less comprehensive.
What about changing this to:

```php
/**
 * @param array $trustedProxies
 * @return void
*/
public static function setTrustedProxies(array $trustedProxies)
 {
    self::$trustedProxiesSettings['proxies'] = $trustedProxies;
}
```
?